src/Explorer/Tabs/DocumentsTabContent.tsx

```diff
@@ -0,0 +1,811 @@
+      tabTitle: this.props.tabTitle(),
+    });
+    return startKey;
+  };
+
+  private setTraceSuccess = (action: number, startKey: number): void => {
+    const { isExecuting, tabTitle } = this.props;
+    TelemetryProcessor.traceSuccess(
+      action,
+      {
+        dataExplorerArea: Areas.Tab,
+        tabTitle: tabTitle(),
+      },
+      startKey
+    );
+    isExecuting(false);
+  };
+
+  private setTraceFail = (action: number, startKey: number, error: Error | string): void => {
+    const { isExecuting, tabTitle, isExecutionError } = this.props;
+    TelemetryProcessor.traceFailure(
+      action,
+      {
+        dataExplorerArea: Areas.Tab,
+        tabTitle: tabTitle(),
+        error: getErrorMessage(error),
+        errorStack: getErrorStack(error),
+      },
+      startKey
+    );
+    isExecuting(false);
+    isExecutionError(true);
+  };
+
+  private onRevertNewDocumentClick = () => {
+    this.newDocumentButton = tabButtonVisibility(true, true);
+    this.saveNewDocumentButton = tabButtonVisibility(true, false);
+    this.discardNewDocumentChangesButton = tabButtonVisibility(true, false);
+
+    this.updateTabButton();
+    this.setState({
+      isEditorVisible: false,
+      isEditorContentEdited: false,
+    });
+  };
+
+  private onRenderCell = (item: { value: string }): JSX.Element => {
+    return (
+      <div
+        className="documentTabSuggestions"
+        onClick={() =>
+          this.setState({
+            filter: item.value,
+            isSuggestionVisible: false,
+          })
+        }
+      >
+        <Text>{item.value}</Text>
+      </div>
+    );
+  };
+
+  private handleLoadMoreDocument = (): void => {
+    userContext.apiType === "Mongo" ? this.queryDocumentsData() : this.querySqlDocumentsData();
+    this.setState({
+      isSuggestionVisible: false,
+      isAllDocumentsVisible: true,
+    });
+  };
+
+  private updateTabButton = (): void => {
+    useCommandBar.getState().setContextButtons(this.getTabsButtons());
+  };
+
+  private getKey(item: DocumentId): string {
+    return item.rid;
+  }
+
+  private handleDocumentContentChange = (newContent: string): void => {
+    if (this.saveExisitingDocumentButton.visible) {
+      this.saveExisitingDocumentButton = tabButtonVisibility(true, true);
+      this.discardExisitingDocumentChangesButton = tabButtonVisibility(true, true);
+    }
+
+    this.setState(
+      {
+        documentContent: newContent,
+        isEditorContentEdited: true,
+      },
+      () => {
+        this.updateTabButton();
+      }
+    );
+  };
+
+  public render(): JSX.Element {
+    const {
+      columns,
+      isCompactMode,
+      isSuggestionVisible,
+      filter,
+      isFilterOptionVisible,
+      isEditorVisible,
+      documentContent,
+      documentIds,
+      documentSqlIds,
+      editorKey,
+      isAllDocumentsVisible,
+    } = this.state;
+
+    const isPreferredApiMongoDB = userContext.apiType === "Mongo";
+
+    return (
+      <div>
```

Comment:
Do you need this `<div>`? If you need to group elements for the tsx syntax to work, use `<></>` which doesn't add new dom elements when rendered by React.